AI/src/main/java/dk/gruppeseks/bodtrd/ai/AIProvider.java

```diff
@@ -5,19 +5,167 @@
  */
 package dk.gruppeseks.bodtrd.ai;
 
+import dk.gruppeseks.bodtrd.common.data.World;
 import dk.gruppeseks.bodtrd.common.data.entityelements.Position;
 import dk.gruppeseks.bodtrd.common.services.AISPI;
+import java.util.ArrayList;
+import java.util.List;
+import org.openide.util.lookup.ServiceProvider;
 
 /**
  *
  * @author frede
  */
+@ServiceProvider(service = AISPI.class)
 public class AIProvider implements AISPI
 {
+    boolean[][] _grid;
+
     @Override
-    public Position[] getPath(Position start, Position goal)
+    public Position[] getPath(Position start, Position goal, World world)
+    {
+        //The following is dummy-code, untill the map is fixed
+        boolean[][] testGrid =
+        {
+            {
+                true, true, true, true
+            },
+            {
+                true, false, true, false
+            },
+            {
+                true, false, true, true
+            },
+            {
+                true, true, false, true
+            }
+        };
+        world.getGameData().setGrid(testGrid);
+        world.getGameData().setCellsize(50);
+        //End of dummy-code
+
+        _grid = world.getGameData().getGrid();
+        int cellSize = world.getGameData().getCellsize();
+
+        int startX = (int)(start.getX() / cellSize);
+        int startY = (int)(start.getY() / cellSize);
+
+        int goalX = (int)(goal.getX() / cellSize);
+        int goalY = (int)(goal.getY() / cellSize);
+
+        Node endNode = AStarSearch(startX, startY, goalX, goalY);
```

Comment:
IMO the search algorithm should return the path, not the end node. the caller shouldnt have to call anything related to the implementation outside of the scope of the algorithm.
Maybe let it return List<Node> or even List<Position>